### PR TITLE
Feature/cookie consent

### PR DESCRIPTION
**Pull request by alkali47**

### CODE REVIEW

**Comment by jouni** [comment]
@alkali47, please sign the Contributor License Agreement (CLA).

**Comment by alkali47** [comment]
Screen Shot 2021-01-27 at 16 17 54

Can't seem to sign the CLA, no agree button or anything like that

File: articles/ds/components/cookie-consent/index.asciidoc [hunk 1]
```diff
@@ -26,3 +27,97 @@ The usage examples for this component are currently in progress. In the meantime
...
+include::../../../../src/main/java/com/vaadin/demo/component/cookieconsent/CookieConsentBasic.java[render,tags=snippet,indent=0,group=Java]
+----
+
+--
+
+By default, the banner is shown at the top of the screen with a predefined text, a link to https://www.cookiesandyou.com[cookiesandyou.com] which explains what cookies are, and a consent button.
+
+== Localization
+
+Cookie Consent is fully customisable.
```
Comment (jouni) [inline articles/ds/components/cookie-consent/index.asciidoc:50]:
```suggestion
Cookie Consent is fully customizable.
```
Use en_US

File: articles/ds/components/cookie-consent/index.asciidoc [hunk 2]
```diff
@@ -26,3 +27,97 @@ The usage examples for this component are currently in progress. In the meantime
...
+You can customise the message, the "Learn More" link, the "Dismiss" button, as well as the component’s position.
```
Comment (jouni) [inline articles/ds/components/cookie-consent/index.asciidoc:51]:
```suggestion
You can customize the message, the "Learn More" link, the "Dismiss" button, as well as the component’s position.
```

File: articles/ds/components/cookie-consent/index.asciidoc [hunk 3]
```diff
@@ -26,3 +27,97 @@ The usage examples for this component are currently in progress. In the meantime
...
+    display: block;
+  }
+
+  .align-center {
+    text-align: center;
+    display: block;
+  }
+</style>
+++++
+
```
Comment (jouni) [inline articles/ds/components/cookie-consent/index.asciidoc:85]:
```suggestion
```

File: articles/ds/components/cookie-consent/index.asciidoc [hunk 4]
```diff
@@ -26,3 +27,97 @@ The usage examples for this component are currently in progress. In the meantime
...
+|[align-right]#Bottom right#
+|===
+
+[cols="1"]
+|===
+
+|[align-center]#Top (stretch, default)#
+
+|[align-center]#Bottom (stretch)#
+|===
```
Comment (jouni) [inline articles/ds/components/cookie-consent/index.asciidoc:102]: Use built-in Asciidoc table cell alignment features.

```suggestion
[cols="<,>"]
|===
|Top left
|Top right

|Bottom left
|Bottom right
|===

[cols="^"]
|===
|Top (stretch, default)
|Bottom (stretch)
|===
```

File: articles/ds/components/cookie-consent/index.asciidoc [hunk 5]
```diff
@@ -26,3 +27,97 @@ The usage examples for this component are currently in progress. In the meantime
 - https://vaadin.com/components/vaadin-cookie-consent/html-examples[Web Component Examples]
 - https://vaadin.com/components/vaadin-cookie-consent/java-examples[Java Component Examples]
 ====
+
+[.example]
+--
+
+[source,typescript]
+----
+include::../../../../frontend/demo/component/cookieconsent/cookie-consent-basic.ts[render,tags=snippet,indent=0,group=TypeScript]
```
Comment (jouni) [inline articles/ds/components/cookie-consent/index.asciidoc:36]: Change this for all code example includes.

```suggestion
include::{root}/frontend/demo/component/cookieconsent/cookie-consent-basic.ts[render,tags=snippet,indent=0,group=TypeScript]
```

**Review by jouni — CHANGES_REQUESTED** [review]

**Review by jouni — APPROVED** [review]

**Pull request merged**